blue_green_deploy.go

```diff
@@ -48,8 +55,39 @@ func (p *BlueGreenDeploy) DeleteAllAppsExceptLiveApp(appName string) {
 
 }
 
-func (p *BlueGreenDeploy) PushNewApp(appName string, route plugin_models.GetApp_RouteSummary, manifestPath string) {
+func (p *BlueGreenDeploy) GetScaleParameters(appName string) AppScaleParameters {
+	appModel, err := p.Connection.GetApp(appName)
+	if err != nil {
+		return AppScaleParameters{}
+	}
+	appScaleParameters := AppScaleParameters{
+		InstanceCount: appModel.InstanceCount,
+		Memory:        appModel.Memory,
+		DiskQuota:     appModel.DiskQuota,
+	}
+	return appScaleParameters
+}
+
+func appendScaleArguments(args []string, scaleParameters AppScaleParameters) []string {
+	if scaleParameters.InstanceCount != 0 {
+		instanceCount := fmt.Sprintf("%d", scaleParameters.InstanceCount)
+		args = append(args, "-i", instanceCount)
+	}
+	if scaleParameters.Memory != 0 {
+		memory := fmt.Sprintf("%dM", scaleParameters.Memory)
+		args = append(args, "-m", memory)
+	}
+	if scaleParameters.DiskQuota != 0 {
+		diskQuota := fmt.Sprintf("%dM", scaleParameters.DiskQuota)
+		args = append(args, "-k", diskQuota)
+	}
+	return args
+}
+
+func (p *BlueGreenDeploy) PushNewApp(appName string, route plugin_models.GetApp_RouteSummary,
+	manifestPath string, scaleParameters AppScaleParameters) {
 	args := []string{"push", appName, "-n", route.Host, "-d", route.Domain.Name}
```

Comment:
That sounds like a good idea. It would be an improvement to raise an error for invalid args to the push call. I would suggest adding this as a separate pull request though.